action.yml

```diff
@@ -39,23 +39,45 @@ inputs:
   checks-to-wait-for:
     description: 'The comma-separated names of the checks to wait for. Only used when wait-for is "ci"'
     required: false
+
 runs:
-  using: "docker"
-  image: "Dockerfile"
-  env:
-    GITHUB_TOKEN: ${{ inputs.token }}
-    GITHUB_CI_CHECKS: ${{ inputs.checks-to-wait-for }}
-    GITHUB_APP_ID: ${{ inputs.app-id }}
-    GITHUB_APP_INSTALLATION_ID: ${{ inputs.app-installation-id }}
-    GITHUB_APP_PRIVATE_KEY: ${{ inputs.app-private-key }}
-  args:
-    - --log-level
-    - debug
-    - --timeout
-    - ${{ inputs.timeout }}
-    - --recheck-interval
-    - ${{ inputs.interval }}
-    - ${{ inputs.wait-for }}
-    - ${{ inputs.owner }}
-    - ${{ inputs.repo }}
-    - ${{ inputs.ref }}
+  using: composite
+
+  steps:
+    - name: Check repository out
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      env:
+        action_repo: ${{ github.action_repository }}
+        action_ref: ${{ github.action_ref }}
+      with:
+        path: ${{ github.workspace }}/action-checkout
+        repository: ${{ env.action_repo }}
+        ref: ${{ env.action_ref }}
+
+    - name: Set up Go
```

Comment:
Would it make sense to build the action and then put it into a cache or is that overkill for now?

Reply:
We could do that as well, yeah, then avoid having to build/setup go at all